rj_gameplay/rj_gameplay/basic_play_selector.py

```diff
@@ -1,89 +1,271 @@
-    defensive_clear,
+    basic_offense,
     defend_restart,
-    restart,
+    defensive_clear,
     kickoff_play,
     penalty_defense,
     penalty_offense,
     prep_penalty_offense,
+    restart,
 )
-from typing import Tuple, Optional
-
-# TODO: Put new plays into the dict properly
-# TODO: Create different dictionaries for different playbooks
-PLAY_DICT = {
-    situations.PrepareKickoff: kickoff_play.PrepareKickoffPlay,
-    situations.Kickoff: basic122.Basic122,
-    situations.DefendKickoff: kickoff_play.DefendKickoffPlay,
-    situations.DefendRestartOffensive: defend_restart.DefendRestart,
-    situations.DefendRestartMidfield: defend_restart.DefendRestart,
-    situations.DefendRestartDefensive: defend_restart.DefendRestart,
-    situations.DefendRestartOffensiveDirect: defend_restart.DefendRestart,
-    situations.DefendRestartMidfieldDirect: defend_restart.DefendRestart,
-    situations.DefendRestartDefensiveDirect: defend_restart.DefendRestart,
-    situations.Clear: defensive_clear.DefensiveClear,
-    situations.DefendClear: defensive_clear.DefensiveClear,
-    situations.DefendGoal: basic_defense.BasicDefense,
-    situations.MidfieldClear: defensive_clear.DefensiveClear,
-    situations.AttackGoal: basic122.Basic122,
-    situations.OffensiveScramble: basic122.Basic122,
-    situations.MidfieldScramble: basic122.Basic122,
-    situations.DefensiveScramble: defensive_clear.DefensiveClear,
-    situations.SaveBall: basic_defense.BasicDefense,
-    situations.SaveShot: basic_defense.BasicDefense,
-    situations.OffensivePileup: basic122.Basic122,
-    situations.MidfieldPileup: basic122.Basic122,
-    situations.DefensivePileup: basic122.Basic122,
-    situations.MidfieldDefendClear: defensive_clear.DefensiveClear,
-    situations.Shootout: penalty_offense.PenaltyOffense,
-    situations.PrepareShootout: prep_penalty_offense.PreparePenaltyOffense,
-    situations.DefendShootout: penalty_defense.PenaltyDefense,
-    situations.PrepareDefendShootout: penalty_defense.PreparePenaltyDefense,
-    situations.Penalty: penalty_offense.PenaltyOffense,
-    situations.PreparePenalty: prep_penalty_offense.PreparePenaltyOffense,
-    situations.DefendPenalty: penalty_defense.PenaltyDefense,
-    situations.PrepareDefendPenalty: penalty_defense.PreparePenaltyDefense,
-    situations.OffensiveKick: restart.RestartPlay,
-    situations.DefensiveKick: restart.RestartPlay,
-    situations.MidfieldKick: restart.RestartPlay,
-    situations.OffensiveKickDirect: restart.DirectRestartPlay,
-    situations.DefensiveKickDirect: restart.DirectRestartPlay,
-    situations.MidfieldKickDirect: restart.DirectRestartPlay,
-    situations.GoalieClear: defensive_clear.DefensiveClear,
-    situations.Stop: defend_restart.DefendRestart,
-}
 
+POSSESS_MIN_DIST = 0.15
+MIN_PASS_SPEED = 0.9
+MIN_NEAR_BALL_DIST = 0.35
+
+
+class BallPos(Enum):
+    """Enum for representing the possession of the ball."""
+
+    OUR_BALL = 1
+    FREE_BALL = 2
+    THEIR_BALL = 3
```

Comment:
Should there not be a contested state? Like if the ball is very close or between an opponent and friendly robot.

Reply:
added